src/boxes/movie.rs

```diff
@@ -59,45 +60,41 @@ impl Mp4Box for Movie {
             tracks: vec![],
             ..Default::default()
         };
+
         let mut index = start + 8; // skip the first 8 bytes that are headers
+                                   // println!("index: {}, len: {}", index, len);
 
-        while index < len {
+        while index < start + len {
```

Comment:
I'm not sure about these changes for all the modules. I know that you need this for `wide`, but not sure about others. You should test this for different files...

Reply:
The reason we are making this change is that the older logic doesn't work on all type of mp4 files. This is not needed for "wide" atom. This is changed because: the older logic of __ index < len __ does not take into consideration the fact that larger atoms such mdat may come in second or third place. It always expects the index to be smaller than the len which may not be the case when you see a larger-sized atom. The older logic will result in buffer not filled related errors. I've tested this on different mp4 files and this appears to work fine.  Let me know if you come across any mp4 file that doesn't support this.